docs/user/tutorial.rst

```diff
@@ -4,88 +4,611 @@
+    l_hid2_drop = lasagne.layers.DropoutLayer(l_hid2, p=0.5)
+
+Finally, we'll add the fully-connected output layer. The main difference is
+that it uses the softmax nonlinearity, as we're planning to solve a 10-class
+classification problem with this network.
+
+.. code-block:: python
+
+    l_out = lasagne.layers.DenseLayer(
+            l_hid2_drop, num_units=10,
+            nonlinearity=lasagne.nonlinearities.softmax)
+
+As mentioned above, each layer is linked to its incoming layer(s), so we only
+need the output layer(s) to access a network in Lasagne:
+
+.. code-block:: python
+
+    return l_out
+
+
+Custom MLP
+^^^^^^^^^^
+
+The second function has a slightly more extensive signature:
+
+.. code-block:: python
+
+    def build_custom_mlp(input_var=None, depth=2, width=800, drop_input=.2,
+                         drop_hidden=.5):
+
+By default, it creates the same network as ``build_mlp()`` described above, but
+it can be customized with respect to the number and size of hidden layers, as
+well as the amount of input and hidden dropout. This demonstrates how creating
+a network in Python code can be a lot more flexible than a configuration file.
+See for yourself:
+
+.. code-block:: python
+
+    # Input layer and dropout (with shortcut `dropout` for `DropoutLayer`):
+    layer = lasagne.layers.InputLayer(shape=(None, 1, 28, 28),
+                                      input_var=input_var)
+    if drop_input:
+        layer = lasagne.layers.dropout(layer, p=drop_input)
+    # Hidden layers and dropout:
+    nonlin = lasagne.nonlinearities.rectify
+    for _ in range(depth):
+        layer = lasagne.layers.DenseLayer(layer, width, nonlinearity=nonlin)
+        if drop_hidden:
+            layer = lasagne.layers.dropout(layer, p=drop_hidden)
+    # Output layer:
+    softmax = lasagne.nonlinearities.softmax
+    layer = lasagne.layers.DenseLayer(layer, 10, nonlinearity=softmax)
+    return layer
+
+With two ``if`` clauses and a ``for`` loop, this network definition allows
+varying the architecture in a way that would be impossible for a ``.yaml`` file
+in `Pylearn2`_ or a ``.cfg`` file in `cuda-convnet`_.
+
+Note that to make the code easier, all the layers are just called ``layer``
+here -- there is no need to give them different names if all we return is the
+last one we created anyway; we just used different names before for clarity.
+
+
+Convolutional Neural Network (CNN)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Finally, the ``build_cnn()`` function creates a CNN of two convolution and
+pooling stages, a fully-connected hidden layer and a fully-connected output
+layer.
+The function begins like the others:
+
+.. code-block:: python
+
+    def build_cnn(input_var=None):
+        layer = lasagne.layers.InputLayer(shape=(None, 1, 28, 28),
+                                          input_var=input_var)
+
+We don't apply dropout to the inputs, as this tends to work less well for
+convolutional layers. Besides, instead of a :class:`DenseLayer
```

Comment:
I'd remove 'Besides' here. Also, might be a little confusing using both 'kernels' and 'filters'.